homeassistant/components/unifiled/light.py

```diff
@@ -0,0 +1,106 @@
+"""Support for Unifi Led lights."""
+import logging
+
+import voluptuous as vol
+
+from unifiled import unifiled
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    PLATFORM_SCHEMA,
+    SUPPORT_BRIGHTNESS,
+    Light,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_PORT, default="20443"): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Unifi LED platform."""
+
+    # Assign configuration variables.
+    # The configuration check takes care they are present.
+    _ip = config[CONF_HOST]
+    _port = config[CONF_PORT]
+    _username = config[CONF_USERNAME]
+    _password = config[CONF_PASSWORD]
+
+    api = unifiled(_ip, _port, username=_username, password=_password)
+
+    # Verify that passed in configuration works
+    if not api.getloginstate():
+        _LOGGER.error("Could not connect to unifiled controller")
```

Comment:
Should this not `PlatformNotReady`?

Reply:
I based this of this: https://github.com/home-assistant/example-custom-config/blob/master/custom_components/example_light/light.py
It is not done there and not in any other component i could find so i am not sure on this.

Reply:
I think it is good like this at the moment but when it is converted to use the config-flow this should be changed because the configuration is then confirmed good at setup so ie does not endlessly tries to connect to a incorrect server.

Reply:
There are two different cases with different preferred solution.

1. Bad credentials or bad config that is not recoverable unless the user makes an adjustment should result in error log and failure to set up integration.
2. Bad connection due to temporary service outrage should result in a retry of integration setup later, via `ConfigEntryNotReady` or `PlatformNotReady`.

Reply:
With the settings in the config file i can not detect if the host is correct or not so i am leaving it like this for now then. I have the config flow code written and with that i can be sure that the stored config is correct so i can then make it retry.